crates/router/src/connector/threedsecureio/transformers.rs

```diff
@@ -0,0 +1,676 @@
+                errors::ConnectorError::MissingRequiredField {
+                    field_name: "sdk_information",
+                },
+            )?),
+            DeviceChannel::Browser => None,
+        };
+        let acquirer_details = authentication_data
+            .acquirer_details
+            .clone()
+            .get_required_value("acquirer_details")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "acquirer_details",
+            })?;
+        let meta: ThreeDSecureIoConnectorMetaData =
+            to_connector_meta(request.authentication_data.1.connector_metadata.clone())?;
+        Ok(Self {
+            ds_start_protocol_version: meta.ds_start_protocol_version.clone(),
+            ds_end_protocol_version: meta.ds_end_protocol_version.clone(),
+            acs_start_protocol_version: meta.acs_start_protocol_version.clone(),
+            acs_end_protocol_version: meta.acs_end_protocol_version.clone(),
+            three_dsserver_trans_id: authentication_data.threeds_server_transaction_id.clone(),
+            acct_number: card_details.card_number.clone(),
+            notification_url: request
+                .return_url
+                .clone()
+                .ok_or(errors::ConnectorError::RequestEncodingFailed)
+                .into_report()
+                .attach_printable("missing return_url")?,
+            three_dscomp_ind: ThreeDSecureIoThreeDsCompletionIndicator::from(
+                request.threeds_method_comp_ind.clone(),
+            ),
+            three_dsrequestor_url: request.three_ds_requestor_url.clone(),
+            acquirer_bin: acquirer_details.acquirer_bin,
+            acquirer_merchant_id: acquirer_details.acquirer_merchant_id,
+            card_expiry_date: card_details.get_expiry_date_as_yymm()?.expose(),
+            bill_addr_city: billing_address
+                .city
+                .clone()
+                .ok_or(errors::ConnectorError::MissingRequiredField {
+                    field_name: "billing_address.address.city",
+                })?
+                .to_string(),
+            bill_addr_country: billing_country.numeric_id().to_string(),
+            bill_addr_line1: billing_address
+                .line1
+                .clone()
+                .ok_or(errors::ConnectorError::MissingRequiredField {
+                    field_name: "billing_address.address.line1",
+                })?
+                .expose()
+                .to_string(),
+            bill_addr_post_code: billing_address
+                .zip
+                .clone()
+                .ok_or(errors::ConnectorError::MissingRequiredField {
+                    field_name: "billing_address.address.zip",
+                })?
+                .expose()
+                .to_string(),
+            bill_addr_state: billing_state.peek().to_string(),
+            three_dsrequestor_authentication_ind: "01".to_string(),
```

Comment:
value always going to be 01? For hard coded values like this please add comments about the significance.